R/waitUntilFinished.R

```diff
@@ -15,11 +17,33 @@
 #' check(job) # should be complete
 #' }
 #' @export
-wait <- function(.Object, sleep.time = gconfig('sleep.time')){
+setGeneric(name="wait",def=function(.Object, sleep.time){standardGeneric("wait")})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "geojob", sleep.time = "numeric"), definition = function(.Object, sleep.time){
+  wait(id(.Object), sleep.time = sleep.time)
+})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "geojob", sleep.time = "missing"), definition = function(.Object, sleep.time){
+  wait(id(.Object), sleep.time = gconfig('sleep.time'))
+})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "character", sleep.time = "numeric"), definition = function(.Object, sleep.time){
   running <- running(.Object, retry = TRUE)
   while(running){
     Sys.sleep(sleep.time)
     running <- running(.Object, retry = TRUE)
   }
   invisible(.Object)
-}
+})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "character", sleep.time = "missing"), definition = function(.Object, sleep.time){
```

Comment:
... but now I see that the pattern I linked to isn't actually using that default because of the way s4 works (which is why I used "missing" here)...oh boy